### PR TITLE
enable dependency track

### PR DESCRIPTION
**Pull request by v-pegao**
[AB#531544](https://dev.azure.com/ceapex/d3d54af3-265a-4f18-95f6-9a46397ca583/_workitems/edit/531544)

### CODE REVIEW

File: src/docfx/lib/log/Telemetry.cs [hunk 1]
```diff
@@ -17,47 +18,75 @@ internal static class Telemetry
...
+    private static readonly AsyncLocal<bool> s_isRealTimeBuild = new();
+
+    private static string s_repo = "<null>";
+    private static string s_branch = "<null>";
+
+    static Telemetry()
+    {
+        using var telemetryConfiguration = TelemetryConfiguration.CreateDefault();
+        using var dependencyTrackingTelemetryModule = new DependencyTrackingTelemetryModule();
+        dependencyTrackingTelemetryModule.Initialize(telemetryConfiguration);
```
Comment (herohua) [inline src/docfx/lib/log/Telemetry.cs:59]: We'll need to sanitize SAS from Azure storage URL. Please check sample code: https://dev.azure.com/ceapex/Engineering/_git/Docs.Build?path=/src/shared/Telemetry.cs&version=GBdevelop&line=21&lineEnd=22&lineStartColumn=1&lineEndColumn=1&lineStyle=plain&_a=contents

File: src/docfx/lib/log/Telemetry.cs [hunk 2]
```diff
@@ -17,47 +18,75 @@ internal static class Telemetry
...
+    private static readonly string s_correlationId = EnvironmentVariable.CorrelationId ?? Guid.NewGuid().ToString("N");
+    private static readonly string s_sessionId = EnvironmentVariable.SessionId ?? "<null>";
+    private static readonly AsyncLocal<bool> s_isRealTimeBuild = new();
+
+    private static string s_repo = "<null>";
+    private static string s_branch = "<null>";
+
+    static Telemetry()
+    {
+        using var telemetryConfiguration = TelemetryConfiguration.CreateDefault();
```
Comment (herohua) [inline src/docfx/lib/log/Telemetry.cs:57]: Can we really dispose this object here (and the line below) as `Telemetry` may still hold reference to it?

**Review by herohua — CHANGES_REQUESTED** [review]

File: src/docfx/docfx.csproj [hunk 3]
```diff
@@ -34,6 +34,7 @@
     <PackageReference Include="Stubble.Core" Version="1.9.3" />
     <PackageReference Include="System.CommandLine" Version="2.0.0-beta1.20574.7" />
     <PackageReference Include="YamlDotNet" Version="11.2.1" />
+    <PackageReference Include="Microsoft.ApplicationInsights.DependencyCollector" Version="2.19.0" />
```
Comment (yufeih) [inline src/docfx/docfx.csproj:37]: Sort these lines a to z

File: src/docfx/lib/log/Telemetry.cs [hunk 4]
```diff
@@ -17,47 +18,75 @@ internal static class Telemetry
     // https://github.com/microsoft/ApplicationInsights-Home/blob/master/EndpointSpecs/Schemas/Bond/EventData.bond#L19
     private const int MaxEventPropertyLength = 8192;
     private const int MaxChildrenLength = 5;
-    private static readonly TelemetryClient s_telemetryClient = new(TelemetryConfiguration.CreateDefault());
+
+    private static TelemetryClient TelemetryClient { get; }
```
Comment (yufeih) [inline src/docfx/lib/log/Telemetry.cs:22]: Why is this turned into properties and not inline initialized as before? If the `TelemetryClient` initialization cannot be done with a single expression, you could use a helper method:
```suggestion
    private static readonly TelemetryClient s_telemetryClient = CreateTelemetryClient();
```

File: src/docfx/lib/log/Telemetry.cs (hunk 1 above)
Comment (v-pegao) [reply src/docfx/lib/log/Telemetry.cs]: Added

File: src/docfx/lib/log/Telemetry.cs (hunk 2 above)
Comment (v-pegao) [reply src/docfx/lib/log/Telemetry.cs]: Use dispose to dispose TelemetryConfiguration and DependencyTrackingTelemetryModule

File: src/docfx/docfx.csproj (hunk 3 above)
Comment (v-pegao) [reply src/docfx/docfx.csproj]: Sorted

File: src/docfx/lib/log/Telemetry.cs (hunk 4 above)
Comment (v-pegao) [reply src/docfx/lib/log/Telemetry.cs]: Updated

File: src/docfx/lib/log/Telemetry.cs [hunk 5]
```diff
@@ -283,6 +301,12 @@ public static void Flush()
         Task.WaitAny(Task.Run(s_telemetryClient.Flush), Task.Delay(10000));
     }
 
+    public static void Dispose()
+    {
```
Comment (herohua) [inline src/docfx/lib/log/Telemetry.cs:305]: I think we can skip this implementation as all resources are static.

Comment (v-pegao) [reply src/docfx/lib/log/Telemetry.cs]: Removed

**Review by herohua — APPROVED** [review]

**Review by yufeih — APPROVED** [review]

**Pull request merged**